### PR TITLE
draft v20.2.0-beta.1 release notes

### LINKED ISSUE #7978: v20.2.0-beta.1 Release Notes
Prep date: 8/17/20
Release date: 8/24/20
SHA: `<add>`
Release Manager: <add>

[Docs To Do](https://cockroachlabs.atlassian.net/wiki/spaces/ENG/pages/73105625/Release+process#Releaseprocess-16.Updatedocs):

- [ ] Run script to generate release notes
- [ ] Update docs site
- [ ] Edit release notes - Additions/changes/omissions to the release notes should be left in this issue's comments
- [ ] Once PR is merged / release notes are live, notify @meagan-g in the #releases Slack channel
- [ ] Email releases@ with link to published release notes
- [ ] Generate release notes issues (cc @lnhsingh)

### PR DESCRIPTION
**Pull request by taroface**
Draft release notes.

- [X] The release highlights for the top-level callout are still TBC.
- [X] ~Didn't find any new `docs-release-note` PRs to highlight.~ Added some docs items to 2nd version of release notes.

Fixes #7978.

### CODE REVIEW

File: releases/v20.2.0-beta.1.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+title: What&#39;s New in v20.2.0-beta.1
+toc: true
+summary: Additions and changes in CockroachDB version v20.2.0-beta.1 since version v20.2.0-alpha.3
+---
+
+## August 31, 2020
+
+In addition to various updates, enhancements, and bug fixes, this first v20.2 beta release includes the following major highlights:
+
+- **Backup schedules:** You can now schedule backups natively in CockroachDB. Previously, it was necessary to create your own scheduler service or cron job to kick off [`BACKUP`](../v20.2/backup.html) jobs. You can now set up a recurring schedule that lets CockroachDB handle scheduling, conflict resolution, and resilience for your backups. To create a schedule for a cluster-level backup that runs a full backup every day with the first full backup taken "now", run `CREATE SCHEDULE test_backup_schedule FOR BACKUP INTO ‘<your-backup-storage-location’ RECURRING ‘@daily’ FULL BACKUP ALWAYS WITH EXPERIMENTAL SCHEDULE OPTIONS first_run=NOW`.
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:11]: Should we restructure this as:
You can now set up a recurring schedule that lets CockroachDB handle scheduling, conflict resolution, and resilience for your backups. Previously, it was necessary to create your own scheduler service or cron job to kick off [`BACKUP`](../v20.2/backup.html) jobs. To create a schedule for a cluster-level backup that runs a full backup every day with the first full backup taken "now", run `CREATE SCHEDULE test_backup_schedule FOR BACKUP INTO ‘<your-backup-storage-location’ RECURRING ‘@daily’ FULL BACKUP ALWAYS WITH EXPERIMENTAL SCHEDULE OPTIONS first_run=NOW`.

File: releases/v20.2.0-beta.1.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+### Security updates
+
+- Certificate revocation is now supported via OSCP and the [cluster setting](../v20.2/cluster-settings.html) `security.ocsp.mode`. This makes a call to the OCSP server on every connection attempt and may wait for up to value specified with the setting `security.ocsp.timeout`. [#53218][#53218]
+- Defining or changing [authentication](../v20.2/authorization.html) principals or their credentials now requires the new `CREATELOGIN` option to be set for the requesting user or one of its roles. This includes setting/removing the LOGIN option (whether the principal can log in), initializing or changing the password of a SQL user, and setting the expiration date for a password. Previously, only the `CREATEROLE` option could perform these changes. The pseudo-option `NOCREATELOGIN` can be used to revoke `CREATELOGIN`. The two predefined `root` and `admin` [roles](../v20.2/authorization.html#default-roles) have the option `CREATELOGIN` set by default. [#50601][#50601]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:54]: nit: LOGIN > `CREATELOGIN`

File: releases/v20.2.0-beta.1.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+- Added the ability to resolve the spatial-backed [builtins](../v20.2/functions-and-operators.html#built-in-functions) in the public schema. For example, `public.st_x` works the same as `st_x`. [#52983][#52983]
+- Implemented `ST_Expand` for the `box2d` type. [#52957][#52957]
+- Added privileges to user-defined types. Users can grant privileges on a type by using [`GRANT {USAGE/GRANT/ALL} ON TYPE <type> TO <users>`](../v20.2/grant.html). Users must have `CREATE` privilege in a database to create a type in that database. Users must have `USAGE` privilege to create an object that depends on a type. Users must have `GRANT` privilege to grant more privileges on the type. Owning a type implicitly gives ALL privileges on the type (`USAGE`,`GRANT`). Users must be the owner of the type to drop the type. Users must be the owner of the type to alter the type. After creating a type, the creator can delegate privileges on that type by [`GRANT`](../v20.2/grant.html)ing any of the `USAGE`/`GRANT`/`ALL` privileges. To allow another user to grant privileges, they must have `GRANT` privilege and the privilege they want to [`GRANT`](../v20.2/grant.html). [#51622][#51622]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:93]: nits: 
- ALL > `ALL`
- Users must be the owner of the type to drop the type. Users must be the owner of the type to alter the type. > Users must be the owner of the type to drop or alter the type.

File: releases/v20.2.0-beta.1.md [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+- [`EXPLAIN`](../v20.2/explain.html) now shows fewer "project" nodes. [#52865][#52865]
+- Improved display of spans in [`EXPLAIN`](../v20.2/explain.html). [#52865][#52865]
+- Added support for the `DROP SCHEMA` command. [#52726][#52726]
+- Adds a new [cluster setting](../v20.2/cluster-settings.html) `sql.log.slow_query.internal` which, when turned on in conjunction with the [slow query log](../v20.2/query-behavior-troubleshooting.html#using-the-slow-query-log), causes slow internal queries to be logged to a slow internal query log at `cockroach-sql-slow-internal-only.log`. Internal queries are no longer logged to the slow query log. This new setting is opt-in. The default behavior is to not log slow internal queries. [#52377][#52377]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:97]: nit (for consistency): Adds > Added

File: releases/v20.2.0-beta.1.md [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+- Non-admin users with `CREATEROLE` are no longer permitted to drop users with the admin role. [#52881][#52881]
+- Users can now specify which subdirectory they wish to write a [`BACKUP`](../v20.2/backup.html) to, via `BACKUP INTO 'subdir' IN x`. This will write `BACKUP` data to `x/subdir/`. [#53139][#53139]
+- Implemented the geometry [builtin](../v20.2/functions-and-operators.html#built-in-functions) `ST_CoordDim` as an alias for `ST_NDims`. [#53193][#53193]
+- Added support for `ALTER SCHEMA <schema> OWNER TO <owner>`. The command changes the owner of a schema. To use the conditions, the following conditions must be met: The user executing the command must be the of the schema. The user executing the command must be a member of the new owner role. The new owner role must have `CREATE` on the database the schema belongs to. [#52781][#52781]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:117]: nit (Missing word): The user executing the command must be the of the schema > The user executing the command must be the owner of the schema.

File: releases/v20.2.0-beta.1.md [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+- Implemented the geometry [builtins](../v20.2/functions-and-operators.html#built-in-functions) `ST_IsEmpty` and `ST_IsCollection`. [#53217][#53217]
+- The [vectorized execution engine](../v20.2/vectorized-execution.html) now natively supports ordered aggregation with the `DISTINCT` clause. [#53145][#53145]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:119]: nit: Remove "natively"

File: releases/v20.2.0-beta.1.md [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+- Previously, `selectivity` information would be included in [`EXPLAIN ANALYZE`](../v20.2/explain-analyze.html) diagrams if a query was executed via the [vectorized execution engine](../v20.2/vectorized-execution.html). This has been removed due to being confusing and probably unhelpful. [#53153][#53153]
+- `stall time` has been renamed to `IO time` in [`EXPLAIN ANALYZE`](../v20.2/explain-analyze.html) diagrams for queries executed via the [vectorized execution engine](../v20.2/vectorized-execution.html). [#53153][#53153]
+- Implemented geometry [builtin](../v20.2/functions-and-operators.html#built-in-functions) `ST_RemovePoint` [#53244][#53244]
+- An [`INSERT ... ON CONFLICT DO UPDATE`](../v20.2/insert.html#on-conflict-clause) statement without a list of column names after `ON CONFLICT` now results in a SQL syntax error with the error code `42601`. Previously, it errored with the message "there is no unqiue or exlcusion constraint matching the ON CONFLICT specification" and the error code `42P10`. [#53067][#53067]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:134]: nits: 
- unqiue  > unique 
- exlcusion > exclusion

File: releases/v20.2.0-beta.1.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+- Added support for the `WITH DATA` and `WITH NO DATA` arguments for the `REFRESH MATERIALIZED VIEW` command. [#53052][#53052]
+- Added support for the `GRANT ... ON SCHEMA command`. [#53344][#53344]
+- Added the `SHOW TYPES` command to list all user-defined types. [#53386][#53386]
+- Implemented the `ST_IsClosed`, `ST_IsSimple`, and `ST_IsRing` [builtins](../v20.2/functions-and-operators.html#built-in-functions). [#53240][#53240]
+
+### Command-line changes
+
+- The --storage-engine CLI flag no longer defaults to using the last-used storage engine when unspecified. Instead, pebble is used by default unless otherwise specified using the --storage-engine flag. [#52944][#52944]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:142]: nit: --storage-engine > `--storage-engine`

File: releases/v20.2.0-beta.1.md [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+- Adds alias commands `ls` and `rm` for `userfile list` and `userfile delete`. [#53074][#53074]
+- The new debug command `decode-proto` reads descriptor from stdin in hex or base64 format (auto-detected) and a flag --schema=<fully qualified name to decode> with default value `cockroach.sql.sqlbase.Descriptor` and outputs to stdout the deserialized proto in JSON format. If the input is not a hex/base64 encoded proto then it is outputted verbatim. [#52972][#52972]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:144]: nit: --schema=<fully qualified name to decode> >`--schema=<fully qualified name to decode>`

File: releases/v20.2.0-beta.1.md [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+- The `cockroach start` and `start-single-node` commands now enable `--redactable-logs` by default. It is also enabled by default in `cockroach demo` if `--log-dir` is passed. This causes log files to become redactable, so that `cockroach debug merge-log --redact` or `cockroach debug zip --redact` can remove sensitive information out of log files. (Reminder: `cockroach debug zip --redact` only affects *log files*; other items collected by the command can still contain sensitive information) [#53263][#53263]
+- The command-line flag `--socket` has been removed. It was deprecated since v20.1. Use `--socket-dir` in replacement. [#53405][#53405]
+- The command-line `--insecure` has been marked as deprecated. See https://go.crdb.dev/issue/53404 for details. The flag will be removed in a later version in a staged fashion: first, additional security mechanisms will be added to enable more flexible deployments which were previously done using `--insecure`; then the flaf will be removed from server commands, then finally in a later version also from client commands. [#53405][#53405]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:147]: nits: 
- See https://go.crdb.dev/issue/53404 for details > See  issue [#53404](https://go.crdb.dev/issue/53404) for details
- flaf > flag

File: releases/v20.2.0-beta.1.md [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+- Recognize the \dT alias for listing user defined [types](../v20.2/data-types.html). [#53386][#53386]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:148]: nit: \dT > `\dT`

**Comment by taroface** [comment]
@Amruta-Ranade Re-requesting your review as the release was restarted with a new SHA, and quite a lot has been added here.

File: releases/v20.2.0-beta.1.md [hunk 12]
```diff
@@ -0,0 +1,520 @@
...
+### Enterprise edition changes
+
+- [`SHOW BACKUP`](../v20.2/show-backup.html) can be used to list backups in a backup collection created by `BACKUP ... INTO`. [#52758][#52758]
+- [`BACKUP`](../v20.2/backup.html)s with `revision_history` now support including UDTs. [#53160][#53160]
+- User-defined schemas now support being backed up with revision history and restored with `AS OF SYSTEM TIME`. [#53241][#53241]
+- Scheduled jobs are now included in [cluster backups](../v20.2/backup.html#backup-a-cluster). [#53203][#53203]
+- As part of making basic [`BACKUP`](../v20.2/backup.html) and [`RESTORE`](../v20.2/restore.html) free to use without an enterprise license, [`SHOW BACKUP`](../v20.2/show-backup.html) and `SHOW BACKUPS IN` no longer require a license. [#53356][#53356]
+- Backups run from 20.2 nodes are no longer run in an auto-appended backup location that was created by 20.1 nodes. [#53762][#53762]
+
+### SQL language changes
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:75]: Suggestion: Can we club the geospatial changes together?

File: releases/v20.2.0-beta.1.md [hunk 13]
```diff
@@ -0,0 +1,520 @@
...
+### Docker image
+
+{% include copy-clipboard.html %}
+~~~shell
+$ docker pull cockroachdb/cockroach-unstable:v20.2.0-beta.1
+~~~
+
+### Backward-incompatible changes
+
+- A change to the on-disk representation for user-defined schemas, enums, and databases was made that is backward-incompatible with 20.2 alpha releases. Only schemas, enums, and databases in the middle of being dropped at the time of the upgrade are affected, and the upgrade may result in the drop never running to completion. If you are upgrading from a 20.2 alpha release to a 20.2 beta release, we advise that you avoid dropping schemas, enums and databases. [#53387][#53387]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:49]: nit: 20.2 > v20.2 for consistency

File: releases/v20.2.0-beta.1.md [hunk 14]
```diff
@@ -0,0 +1,520 @@
...
+- Only a user which already has the `CREATELOGIN` option (either itself or one of its roles) can grant this option or use `NOCREATELOGIN`. [#50601][#50601]
+- Roles that had the `CREATEROLE` privilege prior to upgrading to this version are also automatically granted `CREATELOGIN`. After the upgrade, `CREATELOGIN` is no longer granted automatically. [#50601][#50601]
+
+### General changes
+
+- `Random()` and `gen_random_uuid()` are now supported as default expressions for [`IMPORT`](../v20.2/import.html). [#52247][#52247]
+- [`IMPORT INTO`](../v20.2/import-into.html) is now supported for `DELIMITED` and `PGCOPY` file formats. [#52628][#52628]
+- Computed columns are now supported in the [`IMPORT INTO`](../v20.2/import-into.html) operation. [#51321][#51321]
+- As part of the transition to 20.2, the migration to upgrade schema change jobs started prior to 20.1 is no longer run. [#52968][#52968]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:63]: nit: 20.2 > v20.2 and 20.1 > v20.1

File: releases/v20.2.0-beta.1.md [hunk 15]
```diff
@@ -0,0 +1,520 @@
...
+- In some cases where `AmbiguousResultErrors` were produced under high load, non-ambiguous, retriable errors are now returned. [#53156][#53156]
+
+### Enterprise edition changes
+
+- [`SHOW BACKUP`](../v20.2/show-backup.html) can be used to list backups in a backup collection created by `BACKUP ... INTO`. [#52758][#52758]
+- [`BACKUP`](../v20.2/backup.html)s with `revision_history` now support including UDTs. [#53160][#53160]
+- User-defined schemas now support being backed up with revision history and restored with `AS OF SYSTEM TIME`. [#53241][#53241]
+- Scheduled jobs are now included in [cluster backups](../v20.2/backup.html#backup-a-cluster). [#53203][#53203]
+- As part of making basic [`BACKUP`](../v20.2/backup.html) and [`RESTORE`](../v20.2/restore.html) free to use without an enterprise license, [`SHOW BACKUP`](../v20.2/show-backup.html) and `SHOW BACKUPS IN` no longer require a license. [#53356][#53356]
+- Backups run from 20.2 nodes are no longer run in an auto-appended backup location that was created by 20.1 nodes. [#53762][#53762]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:73]: nit: 20.2 > v20.2 and 20.1 > v20.1

File: releases/v20.2.0-beta.1.md [hunk 16]
```diff
@@ -0,0 +1,520 @@
...
+- The `--geo-libs` flag introduced in v20.2 alpha versions is now renamed to `--spatial-libs`. [#53721][#53721]
+- The CLI no longer prints a blanket `Time` for queries. Instead, if `show_times` is turned on and the server version is 20.2 or later, the CLI prints two separate times: the server execution time and the network latency. [#52233][#52233]
+
+### Admin UI changes
+
+- Improved how [SQL statement plans](../v20.2/admin-ui-statements-page.html#logical-plan) in the Admin UI are populated, using the new `EXPLAIN` infrastructure. [#52956][#52956]
+- Previously, in some cases, the [Execution Stats page](../v20.2/admin-ui-statements-page.html#execution-stats) would show a confusingly high Overhead latency for a statement. This could happen due to multiple statements being parsed together or due to statement execution being retried. To avoid this, we now stop considering the time between when parsing ends and execution begins when determining service latency. [#53846][#53846]
+- Added the Session list and Session detail pages. We now permit session and query termination from the UI. [#51903][#51903]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:202]: nit: Session list > **Session list** and Session detail > **Session detail**

File: releases/v20.2.0-beta.1.md [hunk 17]
```diff
@@ -0,0 +1,520 @@
...
+- The `workload` sub-commands are no longer marked as experimental. [#53691][#53691]
+- The `--geo-libs` flag introduced in v20.2 alpha versions is now renamed to `--spatial-libs`. [#53721][#53721]
+- The CLI no longer prints a blanket `Time` for queries. Instead, if `show_times` is turned on and the server version is 20.2 or later, the CLI prints two separate times: the server execution time and the network latency. [#52233][#52233]
+
+### Admin UI changes
+
+- Improved how [SQL statement plans](../v20.2/admin-ui-statements-page.html#logical-plan) in the Admin UI are populated, using the new `EXPLAIN` infrastructure. [#52956][#52956]
+- Previously, in some cases, the [Execution Stats page](../v20.2/admin-ui-statements-page.html#execution-stats) would show a confusingly high Overhead latency for a statement. This could happen due to multiple statements being parsed together or due to statement execution being retried. To avoid this, we now stop considering the time between when parsing ends and execution begins when determining service latency. [#53846][#53846]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:201]: nit: Overhead > **Overhead** (I am assuming it's a UI component)

File: releases/v20.2.0-beta.1.md [hunk 18]
```diff
@@ -0,0 +1,520 @@
...
+- CockroachDB could previously crash when evaluating queries with window functions with `GROUPS` mode of framing when `OFFSET FOLLOWING` boundary was used and when the offset was a very large value such that it could result in an integer overflow. This is now fixed. [#53722][#53722]
+- Database creation/deletion was previously not correctly tracked by `revision_history` cluster backups. This is now fixed. [#53667][#53667]
+- Fixed a bug that caused a crash when using a `RANGE` mode window function with an offset in some cases, e.g., `OVER (PARTITION BY b ORDER BY a RANGE 1 PRECEDING)`. [#53717][#53717]
+- Fixed a panic in `ST_SetPoint` when given empty point. [#53857][#53857]
+- Fixed a rare internal error that could occur during planning of queries with many highly selective predicates. [#53802][#53802]
+- CockroachDB now properly checks that a user has `CREATE` privileges on the target database before allowing them to create a schema. [#53837][#53837]
+- CockroachDB could previously crash when performing an interleaved join in some cases. This is now fixed. The bug has only been present on 20.2 testing releases. [#53863][#53863]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:231]: nit: 20.2 > v20.2

File: releases/v20.2.0-beta.1.md [hunk 19]
```diff
@@ -0,0 +1,520 @@
...
+- Fixed "use of crdb_internal_vtable_pk column not allowed" for some queries involving virtual tables. [#53866][#53866]
+- Fixed a bug when running restore while [upgrading the cluster from 20.1 to 20.2](../v20.2/upgrade-cockroach-version.html). [#53848][#53848]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:233]: nit: 20.1 > v20.1 and 20.2 > v20.2

File: releases/v20.2.0-beta.1.md [hunk 20]
```diff
@@ -0,0 +1,520 @@
...
+- Transient node liveness blips no longer cause up-to-date ranges to unquiesce, which makes these events less destabilizing. [#51894][#51894]
+- The `~` and `&&` geospatial bounding box operations can now benefit from index acceleration if one of the operands is an indexed geometry column. [#53023][#53023]
+- Cleaning up after a failure during [`IMPORT INTO`](../v20.2/import-into.html) a table which was empty is now faster. [#52754][#52754]
+- The allocation algorithm now supports intra-node rebalances, which means CockroachDB can fully utilize the additional stores on the same node. This removes the last roadblock to running CockroachDB with multiple stores (i.e., disks) per node. [#51567][#51567]
+- Validation of optimistic reads is now performed earlier in transactions when doing so can save work. This eliminates certain types of [transaction retry errors](../v20.2/transaction-retry-error-reference.html) and avoids wasted RPC traffic. [#52884][#52884]
+- Fixes a performance regression introduced in 20.2 alpha releases where the use of 2-part names to specify tables or types would incur extra reads from the store on every transaction. [#53290][#53290]
```
Comment (Amruta-Ranade) [inline releases/v20.2.0-beta.1.md:246]: nit: 20.2 > v20.2

**Review by Amruta-Ranade — APPROVED** [review]

**Comment by taroface** [comment]
TFTR @Amruta-Ranade !

**Pull request merged**